projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -134,3 +136,110 @@ def _start_job(self, node):
                 self._delayed_starting_jobs.enqueue(node)
             else:
                 raise
+
+    def _validate_job(self, job: Dict):
```

Comment:
Also consider moving the logic in separate class: 

 This will follow better the spirit  SRP([Single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle)) and better separation of concerns
 
 DagValidator with public method validate(jobs: List[Dict]) ;

Reply:
Created the suggested new class and moved the logic there.